client/my-sites/media-library/index.jsx

```diff
@@ -107,7 +107,11 @@ class MediaLibrary extends Component {
 	}
 
 	filterRequiresUpgrade() {
-		const { filter, site } = this.props;
+		const { filter, site, source } = this.props;
```

Comment:
This could also be shortened to `{ this.props.source && this.filterRequiresUpgrade() }` in `render()`.